spec/requests/titles_spec.rb

```diff
@@ -36,22 +35,56 @@
       end
     end
 
-    describe 'with filtering' do
+    describe 'filtering by publication type' do
       before do
-        VCR.use_cassette('search-titles-filtered') do
-          get '/eholdings/titles/?q=ebsco&filter=book', headers: okapi_headers
+        VCR.use_cassette('search-titles-filter-book') do
+          filter = { filter: { type: 'book' } }.to_query
+          get "/eholdings/titles/?q=news&#{filter}", headers: okapi_headers
         end
       end
 
       let!(:json_f) { Map JSON.parse response.body }
 
-      it 'gets a filtered list of resources' do
+      it 'gets a list of book resources' do
         expect(response).to have_http_status(200)
-        expect(json_f.data.length).to equal(1)
-        expect(json_f.meta.totalResults).to equal(1)
-        expect(json_f.data.first.id).not_to eql(json.data.first.id)
+        expect(json_f.data.length).to equal(25)
+        expect(json_f.meta.totalResults).to equal(1296)
         expect(json_f.data.first.attributes.publicationType).to eql('Book')
       end
+
+      describe 'filtering further by selected status' do
+        before do
+          VCR.use_cassette('search-titles-filter-book-selection') do
+            filter = { filter: { type: 'book', selected: true } }.to_query
+            get "/eholdings/titles/?q=news&#{filter}", headers: okapi_headers
+          end
+        end
+
+        let!(:json_f2) { Map JSON.parse response.body }
+
+        it 'gets a list of unselected book resources' do
+          expect(response).to have_http_status(200)
+          expect(json_f2.data.length).to equal(25)
+          expect(json_f2.meta.totalResults).to equal(1278)
+          expect(json_f2.data.first.attributes.publicationType).to eql('Book')
+          expect(json_f2.data.first.id).not_to eql(json_f.data.first.id)
```

Comment:
So there is no way to really test that these are selected or not because titles do not have an `isSelected` attribute. But their related customer-resources do, and _this_ is what this filter represents. For example, there are 1296 results for `?q=news&filter[type]=book`, but 1278 are "selected" and ~1230 or so are "not selected". Because a title can contain both selected and not selected customer-resources.